src/main/java/hudson/remoting/Engine.java

```diff
@@ -263,7 +263,11 @@ public synchronized void startEngine() throws IOException {
                 throw new IOException("Cannot find the JAR Cache location");
             }
             LOGGER.log(Level.FINE, "Using standard File System JAR Cache. Root Directory is {0}", jarCacheDirectory);
-            jarCache = new FileSystemJarCache(jarCacheDirectory, true);
+            try {
+                jarCache = new FileSystemJarCache(jarCacheDirectory, true);
+            } catch (IllegalArgumentException ex) {
+                throw new IOException("Failed to initialize FileSystem JAR Cache in " + jarCacheDirectory, ex);
```

Comment:
@oleg-nenashev I don't have experience here, in advance my apologies for my ignorance. Could we use something like `/tmp` by default in case `jarCacheDirectory` has some problem?

When I say `/tmp`, I mean the temporary folder defined in the O.S.

Reply:
@recena We could do that. Default would be a breaking change for Docker images and other things, which rely on the current default location, but it may be applicable as a fallback. 

Good news is that `-workDir` mode changes the location.